File: Class Codes/08-31/logistic1.m
max_iteration=10000;  % max number of iterations
max_period=256;
start_testing=max_iteration-max_period;
numdiv=5000;
tolerance=1e-8;

bfile=fopen('bifurcations_matlab.txt','w');
pfile=fopen('periods_matlab.txt','w');

for riter=0:numdiv-1
    r=1.0+3.0*(riter)/numdiv;
    
    x=0.001;
    xobs=0;
    period=max_period;
    
    for iter=0:max_iteration-1
        x=logistic_calculate(r,x);
    
        if iter==start_testing
            xobs=x;
            fprintf(bfile,'%f',r);
        end
        if iter>start_testing
            fprintf(bfile,',%f',x);
            if abs(x-xobs)<tolerance
                period=iter-start_testing; 
            end
        end
    end
    fprintf(bfile,'\n');
    fprintf(pfile,'%f,%f\n',r,period);
end

function x=logistic_calculate(r,x)
	x=r*x*(1-x);
end

File: Class Codes/08-31/logistic1.py
#  FUNCTION DEFINITION
def logistic_calculate(r,x):
	#r and x are both passed by value, and cannot be updated within the function
	return r*x*(1-x)
	#this returns the new value of x.
# END OF FUNCTION DEFINITION


# PARAMETERS
max_iteration=10000  # how many iterations to do
max_period=256  # largest period
start_testing=max_iteration-max_period	 #how many numbers to print to the screen
numdiv=5000
tolerance=1e-8

bfile=open('bifurcations_python.txt','w')  #where to write bifurcation data
pfile=open('periods_python.txt','w')		#where to write period data

for riter in range(numdiv):
	r=1+3*riter/numdiv
	
	# INITIALIZATION
	x=0.01    # initial value for x
	xobs=0
	period=max_period
	

	for iter in range(max_iteration):
		x=logistic_calculate(r,x)
		
		if iter==start_testing:
			xobs=x
			bfile.write(str(r)) # "str" command converts to a string
		
		if iter>start_testing:
			bfile.write(","+str(x))
			if (abs(x-xobs)<tolerance) and (period==max_period):
				period=iter-start_testing
		
	pfile.write(str(r)+","+str(period));
			
bfile.close();
pfile.close();
			
		
